resource-types/VSCodeRemoteDevelopment/README.md

```diff
@@ -0,0 +1,173 @@
+
+- **The instance has a stable hostname, even throughout recreation**, so the configuration in VS Code (the SSH URI) always stays the same. 
+
+## Installing the resource type
+
+This resource type is built with the [community Typescript plugin for CloudFormation](https://github.com/eduardomourar/cloudformation-cli-typescript-plugin):
+
+```
+pip install git+https://github.com/eduardomourar/cloudformation-cli-typescript-plugin.git@v0.5.0#egg=cloudformation-cli-typescript-plugin
+```
+
+Note that as of this writing, the Typescript plugin does only support cfn cli version 0.1.x. You will get an error during `cfn submit` when you try to use a higher version.
+
+After checking out this repository,
+
+```
+git clone https://github.com/hypescaler/aws-vscode-remote-containers.git
+```
+
+ a complete build and submission can be started with:
+
+```
+# equals "npm install && npm run build && cfn submit --set-default"
+npm run all
+
+```
+
+### Installing - the Docker way
+
+The included Dockerfile has everything to build and submit the resource type. Provided that you are running on MacOS or Linux and have your AWS CLI config in `~/.aws`:
+
+```
+git clone https://github.com/aws-cloudformation/aws-cloudformation-samples.git
+cd aws-cloudformation-samples/resource-types/VSCodeRemoteDevelopment
+docker build . -t dev-container
+SRC=$(pwd) && docker run --rm -it -e SRC=$SRC -v /var/run/docker.sock:/var/run/docker.sock -v $SRC:$SRC -v ~/.aws:/root/.aws dev-container /bin/bash -c "cd \$SRC && eval \$(/home/linuxbrew/.linuxbrew/bin/brew shellenv) && npm run all"
+```
+
+If you want to use a specific AWS profile, you can add `-e AWS_PROFILE=your_profile_name` to the docker command.
+
+## Installing and preparing VS Code
+
+1) You need to have `ssh` installed.
+2) [Install VS Code](https://code.visualstudio.com/docs/setup/setup-overview).
+3) [Install the Remote - Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension.
+
+## Usage
+
+You need to have a [key pair](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html) set up. Put the name of the key pair into the `Keypair` property in `cloudformation/template.json`. The key must be known by your ssh client. 
+
+```
+aws cloudformation create-stack --stack-name myteststack01 --template-body file://cloudformation/template.json
+```
+
+Get the output of the stack to find the SSH URI of your instance:
+
+```
+aws cloudformation describe-stacks --stack-name myteststack01
+
+ ...
+ "Outputs": [
+                {
+                    "OutputKey": "ssh",
+                    "OutputValue": "ssh://ec2-user@ec2-3-122-69-112.eu-central-1.compute.amazonaws.com"
+                }
+            ],
+ ...
+```
+
+Connect to the instance once via your ssh client so that you can accept the fingerprint:
+
+```
+ssh ec2-user@ec2-3-122-69-112.eu-central-1.compute.amazonaws.com
```

Comment:
obfuscate the host:

ssh ec2-user@ec2-example-do-not-use.compute-1.amazonaws.com